components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java

```diff
@@ -462,6 +469,37 @@ protected void setAuthenticator(ApplicationAuthenticator authenticator) {
         }
     }
 
+    @Reference(
+            name = "session.serializer",
+            service = SessionSerializer.class,
+            cardinality = ReferenceCardinality.OPTIONAL,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetSessionSerializer"
+    )
+    protected void setSessionSerializer(SessionSerializer sessionSerializer) {
+
+        SessionSerializer existingSessionSerializer = FrameworkServiceDataHolder.getInstance().getSessionSerializer();
+
+        if (existingSessionSerializer == null) {
+            FrameworkServiceDataHolder.getInstance().setSessionSerializer(sessionSerializer);
+        } else if (existingSessionSerializer instanceof JavaSessionSerializer) {
+            FrameworkServiceDataHolder.getInstance().setSessionSerializer(sessionSerializer);
+            log.warn("Multiple Session Serializers are registered default serializer:"
```

Comment:
We have to improve this log. There are two scenarios we want to log.

1. default `JavaSessionSerializer` overridden with a custom serializer.
2. Multiple custom serializers are registered.